src/Lang/CorePriv/Type.ml

```diff
@@ -32,6 +33,12 @@ let subst_type x stp tp =
 let rec equal : type k. k typ -> k typ -> bool =
   fun tp1 tp2 ->
   match tp1, tp2 with
+  | _, TUVar _
+  | TUVar _, _ ->
+    InterpLib.Error.report ~cls:FatalError
```

Comment:
Same as before. We should ask, what is the philosophy behind raising errors when we encounter unsolved unification variable? Should we always raise an error, or do the best we can (like in e.g. `subeffect` function). I think, for basic functions, like substitution, checking purity, equality, and subtyping we could try to do the best we can: UIDs stored with unification variables makes possible to compare them. But for more complex functions (e.g. `WellTypedInvariant` module) we should definitely raise an internal error.